ddspipe_core/src/cpp/communication/dds/DdsBridge.cpp

```diff
@@ -277,6 +282,31 @@ void DdsBridge::add_writers_to_tracks_nts_(
     }
 }
 
+utils::Heritable<DistributedTopic> DdsBridge::create_topic_for_participant_nts_(
+        const std::shared_ptr<IParticipant>& participant) noexcept
+{
+    // Make a copy of the Topic to customize it according to the Participant's configured QoS.
+    utils::Heritable<DistributedTopic> topic = topic_->copy();
+
+    // Impose the Topic QoSs that have been pre-configured on the topic.
+
+    // 1. Manually Configured Topic QoSs.
+    for (const auto& manual_topic : manual_topics_)
+    {
+        const auto& participant_ids = manual_topic->participants.get_value();
+
+        if (participant_ids.empty() || participant_ids.count(participant->id()))
+        {
+            topic->topic_qos.set_qos(manual_topic->topic_qos, utils::FuzzyLevelValues::fuzzy_level_hard);
```

Comment:
So finally we are mixing different manual qos, I though we said to keep only one entirely (first one or the one with highest priority). Well, if we are keeping this option for know, I hope it is well documented cause it's not trivial in my opinion. And please also add a comment for developers, without knowing how fuzzy works I would expect the last set to be the effective one.